### PR TITLE
tmuxPlugins: rename function to build plugin

### PR DESCRIPTION
**Pull request by timstott**
###### Motivation for this change

Make a distinction between `stdenv.mkDerivation` and the helper function to build the plugin.

Refactor only. No changes in derivation inputs/outputs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/tmux-plugins/default.nix [hunk 1]
```diff
@@ -1,7 +1,6 @@
 { lib
 , fetchFromGitHub
 , pkgs
-, reattach-to-user-namespace
```
Comment (timstott) [inline pkgs/misc/tmux-plugins/default.nix:4]: :nail_care: I have collocated inside the plugin derivation.
```diff
- sed -e 's:reattach-to-user-namespace:${reattach-to-user-namespace}/bin/reattach-to-user-namespace:g' -i $target/sensible.tmux
+ sed -e 's:reattach-to-user-namespace:${pkgs.reattach-to-user-namespace}/bin/reattach-to-user-namespace:g' -i $target/sensible.tmux
```

**Review by SuperSandro2000 — APPROVED** [review]
LGTM but who is the tmux maintainer? Maybe we should add them to codeowners.

**Comment by timstott** [comment]
@SuperSandro2000 Thanks. `thammers` and `fpletz` are [listed](https://github.com/NixOS/nixpkgs/blob/ddb1c81c07b5a2e0ed96b023bafe8ff55ce3a996/pkgs/tools/misc/tmux/default.nix#L78) as tmux maintainers. The `tmux-plugins` however is very much a mix of maintainers.

I would be pleased to added as a codeowner for `tmux-plugins` given my involvement.

**Review by arnarg — APPROVED** [review]
I'm only the maintainer of two plugins. But LGTM.

**Pull request merged**